src/popup/components/Modals/ClaimGiftCard.vue

```diff
@@ -0,0 +1,406 @@
+        <span class="gray">
+          {{ $t('modals.claimGiftCard.redeemedBy', [`(${currencyFormatted})`]) }}
+        </span>
+      </div>
+    </template>
+    <DetailsItem>
+      <template #label>
+        <AccountSelector
+          v-if="step !== STEPS.redeemPartially && step !== STEPS.redeemFull"
+          v-model="recipientId"
+          :options="aeAccountsSelectOptions"
+        />
+        <AccountItem
+          v-else
+          :address="recipientId"
+          :protocol="PROTOCOL_AETERNITY"
+        />
+      </template>
+    </DetailsItem>
+    <div class="redeem-balance">
+      <span v-if="step !== STEPS.initial || !isCardEmpty">
+        {{ $t('modals.claimGiftCard.balance') }}
+      </span>
+      <BalanceInfo
+        :balance="balance.toNumber()"
+        :protocol="PROTOCOL_AETERNITY"
+        :class="{
+          gray: balance.toNumber() === 0,
+        }"
+      />
+    </div>
+    <span
+      v-if="isCardEmpty"
+      class="already-redeemed"
+    >
+      {{ $t('modals.claimGiftCard.alreadyRedeemed') }}
+    </span>
+    <TransferSendAmount
+      v-if="step === STEPS.form"
+      v-model="amount"
+      :errors="errors"
+      readonly
+      :protocol="PROTOCOL_AETERNITY"
+      custom-label="Amount to redeem"
+      :validation-rules="{
+        max_redeem: max.toString(),
+      }"
+    >
+      <template #label-after>
+        <BtnPlain
+          class="max-button"
+          :class="{ chosen: amount.toString() === max.toString() }"
+          @click="setMaxAmount"
+        >
+          {{ $t('common.max') }}
+        </BtnPlain>
+      </template>
+    </TransferSendAmount>
+    <template #footer>
+      <div class="buttons">
+        <BtnMain
+          v-if="step === STEPS.initial"
+          variant="muted"
+          :text="$t('modals.claimGiftCard.redeemPartially')"
+          :disabled="loading"
+          @click="step = STEPS.form"
+        />
+        <BtnMain
+          :icon="loading ? AnimatedSpinner : null"
+          :text="mainButtonText"
+          :disabled="isDisabled || loading"
+          @click="handleMainButtonClick"
+        />
+      </div>
+    </template>
+  </Modal>
+</template>
+
+<script lang="ts">
+import BigNumber from 'bignumber.js';
+import {
+  computed, defineComponent, ref, PropType, onMounted,
+} from 'vue';
+import { useStore } from 'vuex';
+import { useForm } from 'vee-validate';
+import { useI18n } from 'vue-i18n';
+import {
+  AE_AMOUNT_FORMATS,
+  buildTx,
+  encode,
+  Encoded,
+  Encoding,
+  Tag,
+  unpackTx,
+} from '@aeternity/aepp-sdk';
+
+import {
+  useAccounts,
+  useAeSdk,
+  useCurrencies,
+} from '@/composables';
+import { AE_COIN_PRECISION } from '@/protocols/aeternity/config';
+import { getAccountFromSecret } from '@/protocols/aeternity/helpers';
+import { PROTOCOL_AETERNITY } from '@/constants';
+import CheckCircleIcon from '@/icons/check-circle.svg?vue-component';
+import type { RejectCallback, ResolveCallback, ObjectValues } from '../../../types';
```

Comment:
I suggest to put the `types` import above all local imports and use the `@` in the import path.